src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Serialization/ObjectWriters.cs

```diff
@@ -92,6 +92,32 @@ public static void WriteProperties(JsonDataWriter writer, TagHelperDescriptor va
         writer.WriteArrayIfNotNullOrEmpty(nameof(value.Diagnostics), value.Diagnostics, Write);
         writer.WriteObject(nameof(value.Metadata), value.Metadata, WriteMetadata);
 
+        static void WriteDocumentationObject(JsonDataWriter writer, string propertyName, object? documentationObject)
+        {
+            switch (documentationObject)
+            {
+                case DocumentationDescriptor descriptor:
+                    writer.WriteObject(propertyName, descriptor, static (writer, value) =>
+                    {
+                        writer.Write(nameof(value.Id), (int)value.Id);
+                        if (value.Args is { Length: > 0 })
+                        {
+                            writer.WriteArray(nameof(value.Args), value.Args, static (w, v) => w.WriteValue(v));
+                        }
+                    });
+
+                    break;
+
+                case string text:
+                    writer.Write(propertyName, text);
+                    break;
+
+                case null:
+                    // Don't write anything if there isn't any documentation.
+                    break;
```

Comment:
Can we add `default: Debug.Assert(false);` to catch if more types are possible in the future?